src/main/java/org/spongepowered/api/world/extent/StorageType.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.world.extent;
+
+/**
+ * Storage types that can be used for storing blocks and biomes.
+ * These are used when obtaining new copies. Storage obtained
+ * from the implementation should be assumed to be {@link #STANDARD}
+ * unless specified otherwise.
+ */
+public enum StorageType {
```

Comment:
I was thinking about porting over Spout's compressed atomic storage, so I might be adding `COMPRESSED` and `COMPRESSED_THREAD_SAFE` to the types. This is why I opted for a more generic name.



> On Jul 13, 2015, at 9:46 PM, Chris Sanders <notifications@github.com> wrote:
> 
> In src/main/java/org/spongepowered/api/world/extent/StorageType.java:
> 
> > + * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
> > + * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
> > + * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
> > + * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
> > + * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
> > + * THE SOFTWARE.
> > + */
> > +package org.spongepowered.api.world.extent;
> > +
> > +/**
> > + * Storage types that can be used for storing blocks and biomes.
> > + * These are used when obtaining new copies. Storage obtained
> > + * from the implementation should be assumed to be {@link #STANDARD}
> > + * unless specified otherwise.
> > + */
> > +public enum StorageType {
> Unsure about the name choice here. When I think "StorageType", I think the mechanism of how it is storing (FlatFile, MySQL, etc) not essentially when its safe to do so.
> 
> Naming hasn't ever been...a strong attribute of mine so I have no alternatives for you.
> 
> —
> Reply to this email directly or view it on GitHub.
>